### ISSUE TITLE
email notification not working

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#451]
In GitLab by @etiennesky on Nov 21, 2019, 07:43

Hi @dbeltrankyl since about 2 weeks ago email notifications have stopped working (or rarely happen).

This happens with expids a249, a2dm and a2dn using AS autosubmit/3.12.0b-foss-2015a-Python-2.7.9 .

It used to work fine using the same autosubmit.conf and jobs.conf settings.

**Comment by kinow** [comment]
In GitLab by @etiennesky on Nov 21, 2019, 07:51

in /esarchive/autosubmit/a2dm/tmp/20191119_171625_run.log there is mention that a SIM job has failed, yet no emails were sent, I have notifications enabled for RUNNING FAILED COMPLETED
 and never receive any emails.

I tried running on both AS machines, same problem. thanks!

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Nov 21, 2019, 08:50

Thanks for the report, I'll look at it

**Comment by kinow** [comment]
In GitLab by @etiennesky on Nov 26, 2019, 24:41

Hi @dbeltrankyl any news on this? I have relied on this feature a lot in the past, and since it is not working now, sometimes the jobs fail during the weekend and I am not made aware. thanks!

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Nov 26, 2019, 09:10

Hi, 

Sorry for the wait, I'm looking at it today, do you were using wrappers before? I think that it is not working for wrapper jobs. 

I tried to use the notification system and it works but it is not implemented on wrappers so I'll working today to solve that

**Comment by kinow** [comment]
In GitLab by @etiennesky on Nov 26, 2019, 09:19

Hi thanks for the input I think you are right! I am not sure it worked with wrappers which I started to use only recently.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Nov 26, 2019, 10:56

Hi @etiennesky , fyi @mcastril , @wuruchi.

I've updated 3.12.0b with enable notifications for wrapped jobs. I did tests with horizontal-vertical and vertical wrappers and works, if something goes unexpectedly wrong just disable notifications and I'll fix as soon as possible.

It searches every inner job that has changed the status and the behavior is the same as if the job wasn't inside the wrapper. 

The status of inner jobs aren't fully recorded queue status is not contemplated per example, but these fixes are planned for 3.12.1 so don't worry if the status pass from submitted to completed/failed
